src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/Models/DesignPattern.java

```diff
@@ -0,0 +1,127 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.Models;
+
+class DesignPattern {
+    private final String name;
+    private final DesignPattern.Type type;
+    private final String code;
+    private final String patternCode;
+    private final String description;
+    private final String example;
+
+    public enum Type {
```

Comment:
Jeżeli chcielibyśmy dodać kiedyś jakąś rodzinę wzorców w przyszłości, to łamiemy OCP. Jeżeli wierzymy, że coś może być dodane w przyszłości to może lepiej będzie użyć czegoś dynamiczniejszego, np. mapy.

Reply:
Nie widzę łamania OCP w dodaniu nowego wpisu do `enum`. Wykorzystanie mapy wg mnie nie jest najlepszym pomysłem. Jest kosztowniejsza w użyciu, dodatkowo i tak trzeba ją w pewnym momencie uzupełnić, prawdopodobnie elementami typu `enum`, więc wracamy do punktu wyjścia, a jedynie zwiększamy coupling i ryzyko powstania błędów w przyszłości. Co więcej, zmiana/usunięcie wpisu z `enum` spowoduje błąd kompilacji, a przy użyciu mapy błąd zostanie wykryty dopiero w trakcie działania programu (tzw. runtime)